### PR TITLE
Don't validate imports in `example/` if it contains its own pubspec.yaml

### PR DESCRIPTION
**Pull request by sigurdm**
Proposed fix of: https://github.com/dart-lang/pub/issues/1813

### CODE REVIEW

File: lib/src/validator/strict_dependencies.dart [hunk 1]
```diff
@@ -96,10 +96,15 @@ class StrictDependenciesValidator extends Validator {
...
   /// Validates that no Dart files in `benchmark/`, `example/, `test/` or
   /// `tool/` have dependencies that aren't in [deps] or [devDeps].
+  ///
+  /// `example/` is only validated if it does not contain its own pubspec.
   void _validateBenchmarkExampleTestTool(
       Set<String> deps, Set<String> devDeps) {
-    for (var usage
-        in _usagesBeneath(['benchmark', 'example', 'test', 'tool'])) {
+    var directories = ['benchmark', 'test', 'tool'];
+    if (!fileExists(entrypoint.root.path('example', 'pubspec.yaml'))) {
```
Comment (jonasfj) [inline lib/src/validator/strict_dependencies.dart]: Hmm, it's it better to either care about the contents of the `example/` folder, or not care at all.

Comment (sigurdm) [reply lib/src/validator/strict_dependencies.dart]: Makes sense.
Done

File: lib/src/validator/strict_dependencies.dart [hunk 2]
```diff
@@ -98,8 +98,8 @@ class StrictDependenciesValidator extends Validator {
   /// `tool/` have dependencies that aren't in [deps] or [devDeps].
   void _validateBenchmarkExampleTestTool(
```
Comment (jonasfj) [inline lib/src/validator/strict_dependencies.dart]: We should rename this

Comment (sigurdm) [reply lib/src/validator/strict_dependencies.dart]: done

**Pull request merged**